### PR TITLE
Pypi build and python3 support

### PR DESCRIPTION
**Pull request by vicpopov**
After merge it would be possible to install pyaloha with simple command:
```
pip install "git+https://github.com/mapsme/Alohalytics.git@0.0.1#egg=pyaloha&subdirectory=snippets"
```
Also done some python3 compatibility changes.

### CODE REVIEW

File: snippets/pyaloha/ccode.py [hunk 1]
```diff
@@ -2,11 +2,15 @@
...
 import os
+import sys
 
 from pyaloha.protocol import SerializableDatetime
 
-
-ctypes.set_conversion_mode('utf8', 'strict')
+try:
+    from ctypes import set_conversion_mode
+    set_conversion_mode('utf8', 'strict')
```
Comment (burivuh) [inline snippets/pyaloha/ccode.py]: outside try-except clause pls

Comment (burivuh) [reply snippets/pyaloha/ccode.py]: try-except-else

File: snippets/pyaloha/protocol.py [hunk 2]
```diff
@@ -145,6 +145,8 @@ class CustomEncoder(json.JSONEncoder):
     def default(self, obj):
         if hasattr(obj, '__dumpdict__'):
             return obj.__dumpdict__()
+        if isinstance(obj, bytes):
+            return obj.decode()
```
Comment (burivuh) [inline snippets/pyaloha/protocol.py:149]: encode?

File: snippets/pyaloha/ccode.py [hunk 3]
```diff
@@ -151,10 +155,7 @@ def stripped(self):
 def iterate_events(stream_processor, events_limit):
     base_path = os.path.dirname(os.path.abspath(__file__))
     c_module = ctypes.cdll.LoadLibrary(
-        os.path.join(
-            base_path, '..',
-            'c_api', 'build', 'iterate_events.so'
-        )
+        os.path.join(base_path, 'iterate_events.so')
```
Comment (burivuh) [inline snippets/pyaloha/ccode.py:159]: Please modify Makefile to copy so file to a new location for a legacy usage

Comment (vicpopov) [reply snippets/pyaloha/ccode.py:159]: > Please modify Makefile to copy so file to a new location for a legacy usage

done

File: snippets/pyaloha/ccode.py (hunk 1 above)
Comment (vicpopov) [reply snippets/pyaloha/ccode.py]: >outside try-except clause pls
> try-except-else

done

File: snippets/pyaloha/protocol.py (hunk 2 above)
Comment (vicpopov) [reply snippets/pyaloha/protocol.py:149]: > encode?

no. bytes can be decoded. strings can be encoded with encoding into bytes.

**Review by jbenua — APPROVED** [review]

File: snippets/setup.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+from setuptools import setup, find_packages
+from distutils.command.build_clib import build_clib
+from distutils.command.install_lib import install_lib
+from distutils import log
+
+NAME = "pyaloha"
+DESCRIPTION = "Alohalytics statistics engine python parser"
+URL = "https://github.com/mapsme/Alohalytics"
+EMAIL = "me@alex.bio"
+AUTHOR = "Alexander Zolotarev"
```
Comment (burivuh) [inline snippets/setup.py:16]: haha, you wish!

Comment (biodranik) [reply snippets/setup.py:16]: Huh? Alexander Borsuk :)

Comment (vicpopov) [reply snippets/setup.py:16]: Copied from LICENSE. Have no rights to change it.
PR for changed name if you wish.

**Pull request merged**